how-to/integrate-with-servicenow-basic/client/src/provider.ts

```diff
@@ -0,0 +1,262 @@
+			{
+				label: "Default",
+				default: "dark",
+				palette: {
+					brandPrimary: "#0A76D3",
+					brandSecondary: "#383A40",
+					backgroundPrimary: "#1E1F23"
+				}
+			}
+		]
+	});
+}
+
+/**
+ * Initialize the DOM elements.
+ */
+async function initializeDOM(): Promise<void> {
+	errorStatus = document.querySelector<HTMLDivElement>("#errorStatus");
+	btnConnect = document.querySelector<HTMLButtonElement>("#btnConnect");
+	btnDisconnect = document.querySelector<HTMLButtonElement>("#btnDisconnect");
+	connectionStatus = document.querySelector<HTMLParagraphElement>("#connectionStatus");
+	btnQuery = document.querySelector<HTMLButtonElement>("#btnQuery");
+	txtQuery = document.querySelector<HTMLInputElement>("#txtQuery");
+	tableResults = document.querySelector<HTMLTableElement>("#tableResults");
+	bodyResults = document.querySelector<HTMLTableSectionElement>("#bodyResults");
+	username = document.querySelector<HTMLDivElement>("#username");
+
+	if (
+		errorStatus &&
+		btnConnect &&
+		btnDisconnect &&
+		connectionStatus &&
+		btnQuery &&
+		txtQuery &&
+		tableResults &&
+		bodyResults &&
+		username
+	) {
+		updateConnectionStatus();
+		if (INSTANCE_URL === "" || CLIENT_ID === "") {
+			errorStatus.textContent =
+				"You must complete the INSTANCE_URL and CLIENT_ID in provider.ts before the example will function";
+			btnConnect.disabled = true;
+			return;
+		}
+
+		btnConnect.addEventListener("click", async () => {
+			if (btnConnect && connectionStatus && username && errorStatus) {
+				serviceNowConnection = undefined;
+				updateConnectionStatus();
+
+				try {
+					errorStatus.textContent = "";
+					contacts = [];
+					btnConnect.disabled = true;
+					connectionStatus.textContent = "ServiceNow is connecting";
+					enableLogging();
+					serviceNowConnection = await connect(INSTANCE_URL, CLIENT_ID);
+
+					username.textContent = (serviceNowConnection.currentUser.name as string) ?? "Unknown name";
+				} catch (err) {
+					errorStatus.textContent = `Error connecting to ServiceNow\n${formatError(err)}`;
+				}
+				updateConnectionStatus();
+			}
+		});
+
+		btnDisconnect.addEventListener("click", async () => {
+			try {
+				contacts = [];
+				if (txtQuery) {
+					txtQuery.value = "";
+				}
+				if (serviceNowConnection) {
+					await serviceNowConnection.disconnect();
+				}
+			} catch {
+			} finally {
+				if (username) {
+					username.textContent = "Not connected";
+				}
+			}
+			serviceNowConnection = undefined;
+			updateConnectionStatus();
+		});
+
+		btnQuery.addEventListener("click", async () => {
+			if (btnQuery && txtQuery && errorStatus && serviceNowConnection) {
+				try {
+					btnQuery.disabled = true;
+					txtQuery.disabled = true;
+					errorStatus.textContent = "";
+
+					contacts = [];
+					updateContactsTable();
+
+					const query = `nameLIKE${encodeURIComponent(txtQuery.value)}`;
+
+					const response = await serviceNowConnection.executeApiRequest<ServiceNowEntities.Core.Contact[]>(
```

Comment:
I think in all our integrations workspace starters, wherever we call `executeApiRequest` we should handle `AuthTokenExpiredError` errors. Customer's apps will encounter these errors once refresh tokens expire so it's good practice to always handle them, even if it's just a popup saying "Your ServiceNow session has expired, please re-connect".